### PR TITLE
feat: Add JSON path to stylish formatter

### PR DESCRIPTION
**Pull request by aburgel**
When running the linter against a generated OpenAPI document, line numbers aren't all that helpful in seeing where the source of an issue is coming from. In my case, we generate the doc in a CI build, so don't easily have access to the generated file. We do have access to the CI output, and having a JSON path means we can see what path is producing the error.

So this PR adds the JSON path to the stylish formatter makes it easier to see the cause. 

**Checklist**

- [X] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

**Screenshots**
Screen Shot 2020-10-23 at 3 57 01 PM

### CODE REVIEW

**Comment by aburgel** [comment]
Just realized I didn't update all the scenario tests. There's a lot, so I'll hold off on that until there's some indication that this PR may be merged.

**Comment by P0lip** [comment]
@philsturgeon what do you think about this one?

**Comment by philsturgeon** [comment]
Do we want `,` in there? Is this just javascript turning an array into a string and defaulting to using , or is it intentional?

**Comment by aburgel** [comment]
> Do we want , in there? Is this just javascript turning an array into a string and defaulting to using , or is it intentional?

Yeah, it looks like it's an array getting turned into a string. I'm using `IRuleResult.path` which is of type `JsonPath` from `@stoplight/types`, which is an array of `string | number`.

Looks like there's [a helper function that I can use: `printPath`](https://github.com/stoplightio/spectral/blob/7f110eb59513b027f7e8a24a07806e7eda6c0338/src/utils/printPath.ts#L41). Any preference on the `PrintStyle` to use?

**Comment by aburgel** [comment]
@philsturgeon @P0lip updated to use `printPath`

**Review by philsturgeon — APPROVED** [review]

File: src/cli/formatters/stylish.ts [hunk 1]
```diff
@@ -69,6 +70,7 @@ export const stylish: Formatter = results => {
       getMessageType(result.severity),
       result.code ?? '',
       result.message,
+      printPath(result.path, PrintStyle.Dot),
```
Comment (P0lip) [inline src/cli/formatters/stylish.ts:73]: This style is fine, I think. :+1:

**Review by P0lip — APPROVED** [review]

**Comment by aburgel** [comment]
@philsturgeon @P0lip what's the release process for spectral? i'm realizing now that this got merged into the develop branch, so it didn't get into the most recent release. when does develop get promoted to master and shipped?

i'd love to start using this feature.

**Comment by P0lip** [comment]
@aburgel The change you implemented will be included in the 5.8.0 version. I'm planning to get it out before the end of 2020.

**Comment by P0lip** [comment]
@aburgel it's in! Enjoy.

**Pull request merged**